quesma/model/bucket_aggregations/dateRange.go

```diff
@@ -68,6 +68,28 @@ func (interval DateTimeInterval) EndTimestampToSQL() (sqlSelect model.Expr, sele
 	return nil, false
 }
 
+func (interval DateTimeInterval) ToWhereClause(fieldName string) model.Expr {
+	begin, isBegin := interval.BeginTimestampToSQL()
+	end, isEnd := interval.EndTimestampToSQL()
+
+	if isBegin {
+		begin = model.NewInfixExpr(model.NewColumnRef(fieldName), ">=", begin)
+	}
+	if isEnd {
+		end = model.NewInfixExpr(model.NewColumnRef(fieldName), "<", end)
+	}
+
+	if isBegin && isEnd {
+		return model.NewInfixExpr(begin, "AND", end)
+	} else if isBegin {
+		return begin
+	} else if isEnd {
+		return end
+	} else {
```

Comment:
nitpick: Probably this `else` is not needed here

Reply:
True, though made code safe here.